src/app/[locale]/posts/[category]/[slug]/page.tsx

```diff
@@ -175,18 +179,58 @@ function pageToTableOfContents(page: any): readonly HeadingData[] {
   return data;
 }
 
+function formatDuration(duration: string): string {
+  const match = duration.match(/PT(\d+H)?(\d+M)?(\d+S)?/);
+  if (!match) {
+    return '';
+  }
+  const hours = parseInt(match[1]) || 0;
+  const minutes = parseInt(match[2]) || 0;
+  const totalMinutes = hours * 60 + minutes; // pretvaranje u minute
+  const formattedMinutes = totalMinutes > 0 ? `${totalMinutes}min` : ""; // samo minute
+  return formattedMinutes.trim();
+}
+
+
+function concatenateBodies(blocks: readonly Block[]): string {
+  let fullText = "";
+  blocks.forEach((block) => {
+    if (block.body) {
+      fullText += block.body;
+    }
+  });
+  return fullText;
+}
+
+function calculateReadingTime(text: string): string {
+  const wordsPerMinute = 200;
+  const wordsPerImage = 12;
+  const words = text.trim().split(/\s+/).length;
+  const imageCount = (text.match(/!\[\]/g) || []).length;
+  const wordsWithImages = words + imageCount * wordsPerImage;
+  const decimalMinutes = wordsWithImages / wordsPerMinute;
+
+  const minutes = Math.ceil(decimalMinutes); // zaokruživanje na višu minutu
+  return `${minutes}min`;
+}
+
 export default async function Page({
   params: { slug, locale },
 }: Props): Promise<JSX.Element> {
   try {
     const post = await getPostBySlug(slug, locale);
     const categories = await getCategories(locale);
     const topics = await getTopics(locale);
-
     const category = categories.find((c) => c.id === post.category)!;
 
     let videoId = post.post_type === "video" ? post.video?.id : undefined;
-
+    const fullText = concatenateBodies(post.blocks);
+    let timeToRead;
+    if (post.post_type === "video") {
+      timeToRead = `${formatDuration(post.video?.data?.contentDetails?.duration || '')} watch`;
+    } else {
+      timeToRead = `${calculateReadingTime(fullText)} read`;
```

Comment:
`timeToRead` should be calculated in `cms-scripts` (in fileToPost function)

Reply:
Thanks for the comment, Tsotne, I moved it to fileToPost function.